### PR TITLE
Make comment mention suggestions work

### PR DESCRIPTION
**Pull request by gbalint**
**Description:**
Fill in the `resolveMentionSuggestions` template function, so liveblocks can resolve mentions.
Searching for mentions searches by the name, and returns the user id.
The search relies on the collaborators in the liveblocks room storage.

image

### CODE REVIEW

File: editor/liveblocks.config.ts [hunk 1]
```diff
@@ -134,42 +134,46 @@ export const {
...
-
-    const storage = await room.getStorage()
-
-    const collabs = storage.root.get('collaborators') as LiveObject<{ [userId: string]: User }>
-    if (collabs == null) {
-      return []
-    }
-
-    const users = Object.values(collabs.toObject()).map((u) => u.toObject())
+    const users = await getAllUsersFromRoom(projectIdToRoomId(projectId))
```
Comment (ruggi) [inline editor/liveblocks.config.ts:137]: maybe I'm thinking wrongly about this one - but does this need to be updated when the project Id changes? (e.g. fork)

Comment (gbalint) [reply editor/liveblocks.config.ts:137]: getProjectID() always returns the current id, so this should survive a fork (and after a projectId change the room context is rerendered).

Comment (balazsbajorics) [reply editor/liveblocks.config.ts:137]: btw shouldn't getProjectID use the editorState instead of reading off of window?

**Review by balazsbajorics — APPROVED** [review]

**Review by ruggi — APPROVED** [review]

File: editor/liveblocks.config.ts (hunk 1 above)
Comment (gbalint) [reply editor/liveblocks.config.ts:137]: I can not access editorState from the liveblocks resolveUsers function, only globals (and it is not a react component or hook, so I can not access the useEditorState hook neither)

Comment (gbalint) [reply editor/liveblocks.config.ts:137]: But in the long run we should implement a backend endpoint to retrieve users, and then this whole resolveUsers hack will become unnecessary.

**Pull request merged**